### DISCUSSION TITLE
Onboarding Bot / CLI

### DISCUSSION THREAD

**Discussion by Ketcap** [kamp-us/pano#667] [Brain storming - Fikir Paylaşımı]
We would like to welcome our contributer and give them good experience on running app or finding what they are looking for. For this:

- ChatGPT Bot 
  - Reviewing files and answering in-depth questions with examples to help contributers to do things that easy to figure in the repo
  - Answering generic questions about the repo (can be solved with cli as well)
- CLI
  - Has answers for common questions
  - Helps to setup the application
  - General knowledge base

**Comment by Ketcap** [comment]
For CLI Tool [Commander](https://www.npmjs.com/package/commander)
To handle GitRepo installation [LangChain- GitHub Loader](https://js.langchain.com/docs/modules/data_connection/document_loaders/integrations/web_loaders/github) 
For POC we don't need to install GitHub repo but if we do an interface online it is usefull
@usirin ai.kamp.us ? answers anything about kampus ? can be pricey tho

**Reply by usirin** [reply]
gpt.kamp.us?

i really hate `ai` word, lol :D

**Reply by usirin** [reply]
I've checked the LangChain's GithubLoader, though i am not sure what i should be looking for. @Ketcap What are you imagining here?

**Reply by Ketcap** [reply]
Right now the idea is to solve onboarding questions or issues fast and independently.
1. CLI tool to explain and show and tell
2. GitHub Loader might be helpful for if someone asks "how can i write relay connection" and then bot can find out through all the files we have and explain it to a user.

**Comment by emincanoguz11** [comment]
We can write ChatGPT bot for using Assistant Role in ChatGPT. Also Assistant Role has a Q&A. So that it's important to the creating a Q&A chatbot.
https://platform.openai.com/docs/guides/gpt-best-practices/strategy-write-clear-instructions

**Reply by Ketcap** [reply]
This is already included in the ChatGPT Bot idea <3 Let me know what else we think and how we can improve in general
